core/src/test/java/org/apache/calcite/test/JdbcAdapterTest.java

```diff
@@ -85,6 +85,31 @@
             + "FROM \"foodmart\".\"sales_fact_1998\"");
   }
 
+  /**
+   * Test case for
+   * <a href="https://issues.apache.org/jira/browse/CALCITE-3115">[CALCITE-3115]
+   * Cannot add JdbcRules which have different JdbcConvention
+   * to same VolcanoPlanner's RuleSet.</a>*/
+  @Test public void testUnionPlan2() {
+    CalciteAssert.model(JdbcTest.FOODMART_SCOTT_MODEL)
+        .query("select \"store_name\" from \"foodmart\".\"store\"\n"
```

Comment:
Can you make the query (adding some filter) slightly more complicated to show the effect of more rules. For instance  `"select \"store_name\" from \"foodmart\".\"store\"\n"
            + "where \"store_name\" = 'Store 1'\n"
            + "union all\n"
            + "select ename from SCOTT.emp where sal > 0"`